### PR TITLE
add sign_up, sign_in and change password for user

### PR DESCRIPTION
**Pull request by Skuuukaaa**
also added the bootstrap gem and smtp config for gmail

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,2 +1,31 @@
...
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      flash[:notice] = 'Profile successfully updated'
```
Comment (b0nn1e) [inline app/controllers/users_controller.rb:19]: Pleas, use i18n

File: app/views/devise/passwords/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        <%= f.label :email, "Email" %><br />
+        <%= f.email_field :email, autofocus: true, autocomplete: "email", class: "text-black bg-light" %>
+      </div>
+
+      <div class="actions mt-3">
+        <%= f.submit "Send me reset password instructions", class: "btn btn-default btn-primary"%>
+      </div>
+    </div>
+  <% end %>
+</div>
```
Comment (b0nn1e) [inline app/views/devise/passwords/new.html.erb:20]: New line missing

File: app/views/devise/passwords/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<% content_for(:title, 'Settings') %>
```
Comment (b0nn1e) [inline app/views/devise/passwords/edit.html.erb:1]: Convert to haml or slim. Here and in other views.
See https://github.com/heartcombo/devise/wiki/How-To:-Create-Haml-and-Slim-Views

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -6,11 +6,38 @@
...
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.8.0/jquery.min.js"></script>
+    <script src="https://cdn.jsdelivr.net/npm/social-likes/dist/social-likes.min.js"></script>
+
+    <meta name="viewport" content="width=device-width, initial-scale=1">
   </head>
 
   <body>
+  <nav class="topnav">
+    <a class="navbar-brand" href="#"><%=link_to  "Start Page", static_pages_home_path%></a>
```
Comment (b0nn1e) [inline app/views/layouts/application.html.erb:21]: Move "Start Page" and other to i18n

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -6,11 +6,38 @@
...
+        <li class="nav-item dropdown">
+          <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button"
+             data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+            Organizations
+          </a>
+        </li>
+      </ul>
+    </div>
+      <% if I18n.locale == I18n.default_locale %>
+        <%= link_to "English", { :locale=>'en' } %>
```
Comment (b0nn1e) [inline app/views/layouts/application.html.erb:34]: Use new syntax: `<%= link_to "English", { locale: 'en' } %>`

File: app/views/layouts/static_pages.html.erb [hunk 6]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
```
Comment (b0nn1e) [inline app/views/layouts/static_pages.html.erb:1]: If it's possible, avoid using several layouts

File: config/environments/development.rb [hunk 7]
```diff
@@ -63,6 +63,17 @@
...
-  # config.action_cable.disable_request_forgery_protection = true
+  config.action_mailer.delivery_method = :smtp
+  host = 'localhost:3000' # replace with your own url
+  config.action_mailer.default_url_options = { host: host }
+
+  # SMTP settings for gmail
+  config.action_mailer.smtp_settings = {
+    address: 'smtp.gmail.com',
+    port: 587,
+    user_name: 'mgorb71@gmail.com',
```
Comment (b0nn1e) [inline config/environments/development.rb:74]: Put credential to secrets

File: config/application.yml.sample [hunk 8]
```diff
@@ -0,0 +1,3 @@
+development:
+  GOOGLE_CLIENT_ID: "209716765770-2l5bumj7s635mts1bo2had8bfidpt29g.apps.googleusercontent.com"
```
Comment (b0nn1e) [inline config/application.yml.sample:2]: Move it to credentials too.

File: test/controllers/omniauth_callbacks_controller_test.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
```
Comment (b0nn1e) [inline test/controllers/omniauth_callbacks_controller_test.rb:1]: Remove `test` folder.
Please add Rspec and configure generators https://stackoverflow.com/a/22400845/3753977

**Review by b0nn1e — CHANGES_REQUESTED** [review]

File: db/migrate/20210719092529_add_user_first_and_second_name_to_users.rb [hunk 10]
```diff
@@ -0,0 +1,6 @@
+class AddUserFirstAndSecondNameToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :user_first_name, :string
```
Comment (ih-andrey-andreev) [inline db/migrate/20210719092529_add_user_first_and_second_name_to_users.rb:3]: Mb just "first_name"?

Comment (Skuuukaaa) [reply db/migrate/20210719092529_add_user_first_and_second_name_to_users.rb:3]: ok

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -1,2 +1,31 @@
...
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      flash[:notice] = t('controllers.successfully_updated')
+      # redirect_to user_path(@user)
```
Comment (b0nn1e) [inline app/controllers/users_controller.rb:20]: Uncomment it?

File: app/helpers/omniauth_callbacks_helper.rb [hunk 12]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
```
Comment (b0nn1e) [inline app/helpers/omniauth_callbacks_helper.rb:1]: You can remove empty helpers files. 
Also, you can configure rails generator to skip generate helpers

File: app/mailers/application_mailer.rb [hunk 13]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
+  default from: 'mgorb71@gmail.com'
```
Comment (b0nn1e) [inline app/mailers/application_mailer.rb:4]: Get it from secrets

File: config/environments/production.rb [hunk 14]
```diff
@@ -91,32 +85,25 @@
...
-  # these configuration options.
-  # config.active_record.database_selector = { delay: 2.seconds }
-  # config.active_record.database_resolver = ActiveRecord::Middleware::DatabaseSelector::Resolver
-  # config.active_record.database_resolver_context = ActiveRecord::Middleware::DatabaseSelector::Resolver::Session
+  config.action_mailer.delivery_method = :smtp
+  host = 'localhost:3000' # replace with your own url
+  config.action_mailer.default_url_options = { host: host }
+
+  # SMTP settings for gmail
+  config.action_mailer.smtp_settings = {
```
[the remark this answers is not in the document]
Comment (Skuuukaaa) [reply config/environments/production.rb:101]: Maksim
add rspec and config generators, delete folder test, put credential (smtp settings and googleoauth)
https://github.com/Skuuukaaa/Social_Network/pull/1/commits/b545e30093a4413e41b872224535b0906ac4677a
maybe, you watched the old commit?)

File: app/controllers/users_controller.rb (hunk 11 above)
Comment (Skuuukaaa) [reply app/controllers/users_controller.rb:20]: in the next pull request uncomment it

**Review by b0nn1e — APPROVED** [review]

**Pull request merged**